### PR TITLE
community[minor]: Add support for Couchbase Vector Store 

### PR DESCRIPTION
**Pull request by lokesh-couchbase**
Add support for Vector Store in [Couchbase](http://couchbase.com/), a distributed NoSQL database.

- Includes Full Tech Search Based Vector Search
- Bump couchbase version in langchain for document loader

Dependencies:
Added the Couchbase SDK as an optional dependency to langchain community.

### CODE REVIEW

**Comment by jacoblee93** [comment]
Is this still draft?

**Comment by lokesh-couchbase** [comment]
> Is this still draft?

Yes, it was draft due to a new Couchbase SDK Release as I had to test that out. It's done and ready for review now

File: langchain/package.json [hunk 1]
```diff
@@ -1255,7 +1255,7 @@
     "cheerio": "^1.0.0-rc.12",
     "chromadb": "^1.5.3",
     "convex": "^1.3.1",
-    "couchbase": "^4.2.10",
+    "couchbase": "^4.3.0",
```
Comment (jacoblee93) [inline langchain/package.json:1258]: Please remove from deps here

File: libs/langchain-community/src/vectorstores/tests/couchbase.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+import { Cluster } from "couchbase";
+import { OpenAIEmbeddings } from "@langchain/openai";
+import { Document } from "@langchain/core/documents";
+import {
+  CouchbaseVectorStore,
+  CouchbaseVectorStoreArgs,
+} from "../couchbase.js";
+
+describe.skip("Couchbase vector store", () => {
+  const connectionString = process.env.DB_CONN_STR ?? "couchbase://localhost";
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/tests/couchbase.int.test.ts:13]: nit: prefix these env vars with `COUCHBASE_`

File: examples/src/indexes/vector_stores/couchbase/.env.example [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Couchbase connection params
+DB_CONN_STR=
```
Comment (jacoblee93) [inline examples/src/indexes/vector_stores/couchbase/.env.example:2]: nit: Prefix with `COUCHBASE_`

File: examples/src/indexes/vector_stores/couchbase/similaritySearch.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import { OpenAIEmbeddings } from "@langchain/openai";
+import {
+  CouchbaseVectorStoreArgs,
+  CouchbaseVectorStore,
+} from "@langchain/community/vectorstores/couchbase";
+import { Cluster } from "couchbase";
+import { TextLoader } from "langchain/document_loaders/fs/text";
+import { CharacterTextSplitter } from "langchain/text_splitter";
+
+const connectionString = process.env.DB_CONN_STR ?? "couchbase://localhost";
```
Comment (jacoblee93) [inline examples/src/indexes/vector_stores/couchbase/similaritySearch.ts:10]: nit: prefix with `COUCHBASE_`

**Comment by jacoblee93** [comment]
Looks good! Small nit if you can change the env var names but otherwise looks good.

**Comment by jacoblee93** [comment]
Changed it myself, thank you!

**Pull request merged**